### DISCUSSION TITLE
403 Error on Step 2 of auth

### DISCUSSION THREAD

**Discussion by agawley** [timdorr/tesla-api#545] [Q&A] [answered]
When executing Step 2 on this page https://tesla-api.timdorr.com/api-basics/authentication I get a 403 error.

Node JS code:

```
axios
  .post(
    "https://auth.tesla.com/oauth2/v3/authorize",
    querystring.stringify({
      _csrf,
      _phase,
      cancel,
      transaction_id,
      identity,
      credential,
    }),
    {
      params: {
        client_id,
        code_challenge,
        code_challenge_method: "S256",
        redirect_uri: "https://auth.tesla.com/void/callback",
        response_type: "code",
        scope: "openid email offline_access",
        state,
      },
    }
  )
  .then((result) => console.log(result.data));
```

**Comment by agawley** [comment, marked as answer]
I suspect this is same as https://github.com/timdorr/tesla-api/discussions/352

**Comment by IMgoRt** [comment]
> When executing Step 2 on this page https://tesla-api.timdorr.com/api-basics/authentication I get a 403 error.
> 
> Node JS code:
> 
> ```
> axios
>   .post(
>     "https://auth.tesla.com/oauth2/v3/authorize",
>     querystring.stringify({
>       _csrf,
>       _phase,
>       cancel,
>       transaction_id,
>       identity,
>       credential,
>     }),
>     {
>       params: {
>         client_id,
>         code_challenge,
>         code_challenge_method: "S256",
>         redirect_uri: "https://auth.tesla.com/void/callback",
>         response_type: "code",
>         scope: "openid email offline_access",
>         state,
>       },
>     }
>   )
>   .then((result) => console.log(result.data));
> ```

Somewhat off-topic, but would you mind sharing your full Node JS code for all 4 steps (and refresh if you have it), thanks in advance
